src/Context/Product-Context/product-context.jsx

```diff
@@ -44,6 +46,17 @@ const ProductProvider = ({ children }) => {
     })();
   }, []);
 
+    const [searchParams] = useSearchParams();
+    const categorygot = searchParams.get("categories");
```

Comment:
suggestion : you could use better naming here, for an example say, selectedCategory.